arm/Microsoft.ApiManagement/service/deploy.bicep

```diff
@@ -224,34 +196,196 @@ resource apiManagementService 'Microsoft.ApiManagement/service@2020-12-01' = {
+    identityProviderSignInPolicyName: contains(identityProvider, 'identityProviderSignInPolicyName') ? identityProvider.identityProviderSignInPolicyName : ''
+    identityProviderSignInTenant: contains(identityProvider, 'identityProviderSignInTenant') ? identityProvider.identityProviderSignInTenant : ''
+    identityProviderSignUpPolicyName: contains(identityProvider, 'identityProviderSignUpPolicyName') ? identityProvider.identityProviderSignUpPolicyName : ''
+    identityProviderType: contains(identityProvider, 'identityProviderType') ? identityProvider.identityProviderType : 'aad'
+  }
+}]
+
+module namedValues_resource 'namedValues/deploy.bicep' = [for (namedValue, index) in namedValues: {
+  name: '${uniqueString(deployment().name, location)}-namedValue-${index}'
+  params: {
+    apiManagementServiceName: apiManagementService.name
+    displayName: namedValue.displayName
+    keyVault: contains(namedValue, 'keyVault') ? namedValue.keyVault : {}
+    name: namedValue.name
+    namedValueTags: contains(namedValue, 'namedValueTags') ? namedValue.namedValueTags : []
+    secret: contains(namedValue, 'secret') ? namedValue.secret : false
+    value: contains(namedValue, 'value') ? namedValue.value : newGuidValue
+  }
+}]
+
+module portalSettings_resource 'portalsettings/deploy.bicep' = [for (portalSetting, index) in portalSettings: {
+  name: '${uniqueString(deployment().name, location)}-portalSetting-${index}'
+  params: {
+    apiManagementServiceName: apiManagementService.name
+    name: portalSetting.name
+    properties: contains(portalSetting, 'properties') ? portalSetting.properties : {}
+  }
+}]
+
+module policy_resource 'policies/deploy.bicep' = [for (policy, index) in policies: {
+  name: '${uniqueString(deployment().name, location)}-policy-${index}'
+  params: {
+    apiManagementServiceName: apiManagementService.name
+    value: policy.value
+    format: contains(policy, 'format') ? policy.format : 'xml'
+  }
+}]
+
+module products_resource 'products/deploy.bicep' = [for (product, index) in products: {
+  name: '${uniqueString(deployment().name, location)}-product-${index}'
+  params: {
+    apiManagementServiceName: apiManagementService.name
+    approvalRequired: contains(product, 'approvalRequired') ? product.approvalRequired : false
+    name: product.name
+    productApis: contains(product, 'productApis') ? product.productApis : []
+    productDescription: contains(product, 'productDescription') ? product.productDescription : ''
+    productGroups: contains(product, 'productGroups') ? product.productGroups : []
+    state: contains(product, 'state') ? product.state : 'published'
+    subscriptionRequired: contains(product, 'subscriptionRequired') ? product.subscriptionRequired : false
+    subscriptionsLimit: contains(product, 'subscriptionsLimit') ? product.subscriptionsLimit : 1
+    terms: contains(product, 'terms') ? product.terms : ''
+  }
+  dependsOn: [
+    apiManagementService
```

Comment:
Remove apiManagementService dependency. It's implicit

Reply:
done